python/lsst/daf/butler/_butler.py

```diff
@@ -1131,6 +1131,43 @@ def getURI(self, datasetRefOrType: Union[DatasetRef, DatasetType, str],
                                "Use Butler.getURIs() instead.")
         return primary
 
+    def getArtifacts(self, refs: Iterable[DatasetRef],
+                     destination: Union[str, ButlerURI], transfer: str = "auto",
+                     preserve_path: Optional[bool] = True) -> List[ButlerURI]:
+        """Retrieve the artifacts associated with the supplied refs.
+
+        Parameters
+        ----------
+        refs : iterable of `DatasetRef`
+            The datasets for which artifacts are to be retrieved.
+            A single ref can result in multiple artifacts. The refs must
```

Comment:
I'm working on the assumption that the normal approach would be `butler.registry.queryDatasets` and then `butler.getArtifacts` on the result, so resolved ref is okay to be required here rather than having to support collection+dataId+datasetType.